python/cable_cell_io.cpp

```diff
@@ -13,7 +13,10 @@
 
 namespace pyarb {
 
-arborio::cable_cell_component load_component(const std::string& fname) {
+namespace py = pybind11;
+
+arborio::cable_cell_component load_component(py::object fn) {
+    const std::string fname = py::str(fn);
```

Comment:
Here we are using the `py::object` as though it can be converted to a `std::string`, without checking.

The errors when a type that can't be converted to string, or converts to an odd string, is passed.

How about writing a little helper function that tests the type and converts it to a string, or throws an exception with a helpful message?

Reply:
... option 1!

Reply:
The Pythonic way is to try and handle errors (which can just be "Unknown input provided, aborting...").